src/main/java/io/jenkins/plugins/autify/AutifyCli.java

```diff
@@ -121,16 +131,37 @@ public void mobileAuthLogin(String mobileAccessToken) {
         this.mobileAccessToken = mobileAccessToken;
     }
 
-    private int runCommand(InputStream stdin, ArgumentListBuilder builder) {
-        return runCommand(stdin, builder.toCommandArray());
+    public Version getVersion() {
+        ByteArrayOutputStream stdout = new ByteArrayOutputStream();
+        int result = execute(new Builder("--version"), stdout);
+        if (result == 0) {
+            Pattern pattern = Pattern.compile("@autifyhq/autify-cli/(.*?) ");
+            String stdoutStr = stdout.toString(Charset.defaultCharset());
+            Matcher matcher = pattern.matcher(stdoutStr);
+            if (matcher.find()) {
+                return Version.parse(matcher.group(1));
+            } else {
+                logger.println(String.format("Failed to parse `%s` as version.", result));
+
+                return null;
+            }
+        } else {
+            logger.println("Failed to execute --version command.");
+
+            return null;
+        }
+    }
+
+    private int runCommand(InputStream stdin, OutputStream stdout, ArgumentListBuilder builder) {
+        return runCommand(stdin, stdout, builder.toCommandArray());
     }
 
-    private int runCommand(InputStream stdin, String... command) {
+    private int runCommand(InputStream stdin, OutputStream stdout, String... command) {
         try {
             ProcStarter procStarter = launcher.launch()
                     .pwd(workspace)
                     .envs(getEnvs())
-                    .stdout(logger)
+                    .stdout(stdout)
```

Comment:
I am asking probably due to lack of familiarity, but could this have undesirable side effects?

Reply:
I believe it doesn't have side effects since even with my change, `runCommand` keeps using `logger` as a argument for `.stdout` except `getVersion`.

https://github.com/jenkinsci/autify-plugin/blob/f58e7b92285e2912313d8757a0985a399943c029/src/main/java/io/jenkins/plugins/autify/AutifyCli.java#L211

https://github.com/jenkinsci/autify-plugin/blob/f58e7b92285e2912313d8757a0985a399943c029/src/main/java/io/jenkins/plugins/autify/AutifyCli.java#L190